CONTRIBUTING.md

```diff
@@ -49,94 +47,101 @@ If you are proposing a feature:
-   ```
-
-8. Submit a pull request through the GitHub website.
-
-## Pull Request Guidelines
-
-Before you submit a pull request, check that it meets these guidelines:
-
-1. The pull request should include tests.
-2. If the pull request adds functionality, the docs should be updated. Put
-   your new functionality into a function with a docstring, and add the
-   feature to the list in README.rst.
-3. The pull request should work for [supported Python versions](https://endoflife.date/python) and for PyPy. Check
-   [GHA](https://github.com/openradar/xradar/actions)
-   and make sure that the tests pass for all supported Python versions.
-
-## Building the documentation
-
-To build the documentation run:
-
-   ```bash
-   $ cd docs
-   $ make html
-   ```
-Then you can access the documentation via browser locally by opening `docs/_build/html/index.html`.
-
-## Tips
-
-To run a subset of tests:
-
-   ```bash
-   $ pytest tests.test_xradar
-   ```
+3. Install `xradar` development environment.
+
+  - Install with conda (Linux):
+
+    ```bash
+    curl -L https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh
+    bash Miniforge3-Linux-x86_64.sh -b
+    rm Miniforge3-Linux-x86_64.sh
+    source $HOME/miniforge3/etc/profile.d/conda.sh
+    conda env create --file environment.yml
+    conda activate xradar-dev
+    ```
+
+  - Install with conda (Windows):
+
+    ```bash
+    winget install -e --id CondaForge.Miniforge3
+    & "$HOME\Miniforge3\shell\condabin\conda-hook.ps1"
+    conda env create --file environment.yml
+    conda activate xradar-dev
+    ```
+
+  - Install with uv (Linux):
+
+    ```bash
+    sudo apt install build-essential ffmpeg libhdf5-dev libnetcdf-dev pandoc
+    sudo snap install astral-uv --classic
+    uv python install 3.13
+    uv venv --python 3.13
+    source .venv/bin/activate
+    uv pip install -e .[dev] --no-binary=netcdf4 --no-binary=h5py
+    ```
+
+  - Install with uv (Windows):
+
+    ```powershell
+    winget install ffmpeg
+    winget install uv
+    uv python install 3.13
+    uv venv --python 3.13
+    .venv\Scripts\Activate.ps1
+    uv pip install -e .[dev]
+    ```
+
+4. Create a branch for local development.
+
+    ```bash
+    git checkout -b name-of-your-bugfix-or-feature
+    ```
+    Now you can make your changes locally.
+
+5. Perform tests and ensure your changes are covered.
+    ```bash
+    python -m pytest -k name_of_my_test
+    python -m pytest -n auto --dist loadfile --cov=xradar --cov-report xml --verbose
+    python -m coverage report
+    diff-cover coverage.xml --compare-branch=main
+    ```
+
+6. Update documentation.
+    - Add docstring to new functions
+        ```bash
+        python -m sphinx build -j auto -v -b html docs/ doc-build
+        ```
+    - Check at `docs/_build/html/index.html`
+    - Add new features to README.rst
+
+7. Run pre-commit hooks
+
+    ```bash
+    git add your-new-file.py
+    pre-commit run --all-files
```

Comment:
Do we need to tell the user to do `pre-commit install` once before running?

Reply:
I would let the developer decides.

Reply:
The guide should be concise. Let's recommend pre-commit install by default !